tests/test-extensions/heisenberg-extension/src/main/java/org/mule/test/heisenberg/extension/HeisenbergScopes.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.test.heisenberg.extension;
+
+import org.mule.runtime.extension.api.annotation.error.Throws;
+import org.mule.runtime.extension.api.annotation.param.Optional;
+import org.mule.runtime.extension.api.annotation.param.stereotype.AllowedStereotypes;
+import org.mule.runtime.extension.api.runtime.process.CompletionCallback;
+import org.mule.runtime.extension.api.runtime.operation.Result;
+import org.mule.runtime.extension.api.runtime.process.Chain;
+import org.mule.test.heisenberg.extension.stereotypes.DrugKillingStereotype;
+import org.mule.test.heisenberg.extension.stereotypes.KillingStereotype;
+
+import java.util.Map;
+
+public class HeisenbergScopes {
+
+  @Throws(HeisenbergErrorTyperProvider.class)
+  public void killMany(@AllowedStereotypes({KillingStereotype.class, DrugKillingStereotype.class}) Chain killOperations,
+                       CompletionCallback<String, Void> callback, String reason)
```

Comment:
not sure if I remember the final decision on this. Do I always need the completionCallback when using chains? Isn't the onSuccess() thing enough?

If I do need the CompletionCAllback, then:

a) That should be clear on the Chain jdoc
b) There should be a validator which fails if I get a chain without a callback
c) all scopes should appear as non blocking in the extension model